### PR TITLE
feat(aws-apigateway): add ability to include authorizer context in apigw sfn integration

### PR DESCRIPTION
**Pull request by rogerchi**
Implements #18891 

This PR adds the ability to include custom authorizer context in the Step Functions API Gateway integration.  This is useful if the custom authorizer sets custom context values, which can then be used downstream in the step function.  This adds a `authorizer: { ... }` key to the state input.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by rogerchi** [comment]
Is there an easy way to generate the expected string from the CFN template that the test is failing on?

**Review by otaviomacedo — DISMISSED** [review]

**Comment by otaviomacedo** [comment]
@rogerchi I updated the tests for you.

File: packages/@aws-cdk/aws-apigateway/README.md [hunk 1]
```diff
@@ -193,6 +193,7 @@ new apigateway.StepFunctionsRestApi(this, 'StepFunctionsRestApi', {
   headers: true,
   path: false,
   querystring: false,
+  authorizer: false,
```
Comment (otaviomacedo) [inline packages/@aws-cdk/aws-apigateway/README.md:196]: This part was actually not implemented. Please take a look at the build failure.

**Review by otaviomacedo — CHANGES_REQUESTED** [review]

File: packages/@aws-cdk/aws-apigateway/README.md (hunk 1 above)
Comment (rogerchi) [reply packages/@aws-cdk/aws-apigateway/README.md:196]: My bad -- I added it to the Integration but not the RestApi

**Review by otaviomacedo — APPROVED** [review]

**Pull request merged**